nhl_api_py/core/models.py

```diff
@@ -135,3 +237,17 @@ def _field_only_keys(data: dict, cls: Type[Model]) -> dict:
     :return: the same dictionary with only the model's fields
     """
     return {k: v for k, v in data.items() if k in (field.name for field in fields(cls))}
+
+
+def _append_string_to_keys(text: str, d: dict) -> dict:
+    """
+    Helper function which appends a string to the top level keys of a dictionary.
+
+    :param text: the text you want to append
+    :param d: the dictionary we want to convert keys for
+    :return: the same dictionary with converted keys
+    """
+    for key, _ in list(d.items()):
```

Comment:
```suggestion
    for key in d:
```
You are extracting items that you are not using.